### PR TITLE
ADIOS2: Update to 2.10.2

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: A/ADIOS2/build_tarballs.jl [hunk 1]
```diff
@@ -60,6 +60,13 @@ else
     archopts+=(-DADIOS2_USE_DataMan=ON -DADIOS2_USE_HDF5=ON -DADIOS2_USE_SST=ON)
 fi
 
+# Use MGARD if it is available
+if [ -e ${prefix}/libmgard.${dlext} ]; then
```
Comment (giordano) [inline A/ADIOS2/build_tarballs.jl:64]: This path looks wrong?

Comment (eschnett) [reply A/ADIOS2/build_tarballs.jl]: Thanks!

File: A/ADIOS2/build_tarballs.jl [hunk 2]
```diff
@@ -168,3 +181,5 @@ ENV["MPITRAMPOLINE_DELAY_INIT"] = "1"
 # GCC 5 is too old for FreeBSD; it doesn't have `std::to_string`
 build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
                augment_platform_block, julia_compat="1.6", preferred_gcc_version=v"6")
+
+# Build trigger: 1
```
Comment (fingolfin) [inline A/ADIOS2/build_tarballs.jl:185]: You don't really need this as you change other stuff in this file (even the version). But of course you can also keep it in, it doesn't harm anything
```suggestion
```

Comment (eschnett) [reply A/ADIOS2/build_tarballs.jl:185]: I needed it earlier because there was a bug in MGARD_jll and I needed to rebuild ADIOS2_jll. I'll remove it again.

**Comment by eschnett** [comment]
I'm giving up on MGARD at the moment, there seems to be a problem in ADIOS2 in this respect that I don't want to address at the moment.

This PR is still useful. It upgrades to a newer version of ADIOS2 and adds support for aarch64-freebsd.

**Pull request merged**